Cannot install under Python 3.10

When attempting to `pip install JPype1` under Windows 10 I get:

```
 C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\bin\HostX86\x64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -DWIN32=1 -Inative\common\include -Inative\python\include -Inative\embedded\include -IC:\Program Files\Java\jdk-17\include -IC:\Program Files\Java\jdk-17\include\win32 -Ic:\users\gili\documents\daikin-one\python\include -IC:\Python310\include -IC:\Python310\Include -IC:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\ATLMFC\include -IC:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include -IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\ucrt -IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\shared -IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\um -IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\winrt -IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\cppwinrt /EHsc /Tpnative\common\jp_array.cpp /Fobuild\temp.win-amd64-3.10\Release\native\common\jp_array.obj /Zi /EHsc /std:c++14
    jp_array.cpp
C:\Users\Gili\AppData\Local\Temp\pip-install-0fjlkma0\jpype1_aff26bff63844fe7b53cb4f7c7e8c054\native\python\include\jp_pythontypes.h(284): error C2061: syntax error: identifier 'ssize_t'
  C:\Users\Gili\AppData\Local\Temp\pip-install-0fjlkma0\jpype1_aff26bff63844fe7b53cb4f7c7e8c054\native\python\include\jp_pythontypes.h(285): error C2805: binary 'operator [' has too few parameters
  C:\Users\Gili\AppData\Local\Temp\pip-install-0fjlkma0\jpype1_aff26bff63844fe7b53cb4f7c7e8c054\native\python\include\jp_pythontypes.h(285): error C2333: 'JPPyObjectVector::operator []': error in function declaration; skipping function body
  C:\Users\Gili\AppData\Local\Temp\pip-install-0fjlkma0\jpype1_aff26bff63844fe7b53cb4f7c7e8c054\native\common\include\jp_primitivetype.h(44): error C3646: 'getItemSize': unknown override specifier
  C:\Users\Gili\AppData\Local\Temp\pip-install-0fjlkma0\jpype1_aff26bff63844fe7b53cb4f7c7e8c054\native\common\include\jp_primitivetype.h(44): error C2059: syntax error: ')'
  C:\Users\Gili\AppData\Local\Temp\pip-install-0fjlkma0\jpype1_aff26bff63844fe7b53cb4f7c7e8c054\native\common\include\jp_primitivetype.h(44): error C2238: unexpected token(s) preceding ';'
  native\common\jp_array.cpp(181): error C2065: 'ssize_t': undeclared identifier
  native\common\jp_array.cpp(181): error C2146: syntax error: missing ';' before identifier 'itemsize'
  native\common\jp_array.cpp(181): error C2065: 'itemsize': undeclared identifier
  native\common\jp_array.cpp(182): error C2065: 'ssize_t': undeclared identifier
  native\common\jp_array.cpp(182): error C2146: syntax error: missing ';' before identifier 'sz'
  native\common\jp_array.cpp(182): error C2065: 'sz': undeclared identifier
  native\common\jp_array.cpp(188): error C2065: 'itemsize': undeclared identifier
  native\common\jp_array.cpp(188): error C2039: 'getItemSize': is not a member of 'JPPrimitiveType'
  C:\Users\Gili\AppData\Local\Temp\pip-install-0fjlkma0\jpype1_aff26bff63844fe7b53cb4f7c7e8c054\native\common\include\jp_primitivetype.h(20): note: see declaration of 'JPPrimitiveType'
  native\common\jp_array.cpp(189): error C2065: 'sz': undeclared identifier
  native\common\jp_array.cpp(189): error C2065: 'itemsize': undeclared identifier
  native\common\jp_array.cpp(193): error C2065: 'sz': undeclared identifier
  native\common\jp_array.cpp(197): error C2065: 'itemsize': undeclared identifier
  native\common\jp_array.cpp(206): error C2065: 'sz': undeclared identifier
  native\common\jp_array.cpp(216): error C2065: 'itemsize': undeclared identifier
  native\common\jp_array.cpp(224): error C2065: 'itemsize': undeclared identifier
  native\common\jp_array.cpp(227): error C2065: 'sz': undeclared identifier
  error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Community\\VC\\Tools\\MSVC\\14.29.30133\\bin\\HostX86\\x64\\cl.exe' failed with exit code 2
```

Python 3.9 used to work just fine.

--- a/.azure/build.yml
+++ b/.azure/build.yml
@@ -22,19 +22,19 @@ variables:
 jobs:
 - job: Deps
   pool:
-    vmImage: "ubuntu-16.04"
+    vmImage: "ubuntu-latest"
   steps:
   - template: scripts/ivy.yml
 
 - job: Documentation
   pool:
-    vmImage: "ubuntu-16.04"
+    vmImage: "ubuntu-latest"
   steps:
   - template: scripts/documentation.yml
 
 - job: Coverage
   pool:
-    vmImage: "ubuntu-16.04"
+    vmImage: "ubuntu-latest"
   dependsOn: Deps
   steps:
   - template: scripts/deps.yml
@@ -42,7 +42,7 @@ jobs:
 
 - job: Tracing
   pool:
-    vmImage: "ubuntu-16.04"
+    vmImage: "ubuntu-latest"
   steps:
   - template: scripts/tracing.yml
 
@@ -51,19 +51,19 @@ jobs:
   strategy:
     matrix:
       linux-3.5:
-        imageName: "ubuntu-16.04"
+        imageName: "ubuntu-latest"
         python.version: '3.5'
       linux-3.6:
-        imageName: "ubuntu-16.04"
+        imageName: "ubuntu-latest"
         python.version: '3.6'
       linux-3.7:
-        imageName: "ubuntu-16.04"
+        imageName: "ubuntu-latest"
         python.version: '3.7'
       linux-3.8:
-        imageName: "ubuntu-16.04"
+        imageName: "ubuntu-latest"
         python.version: '3.8'
       linux-3.9:
-        imageName: "ubuntu-16.04"
+        imageName: "ubuntu-latest"
         python.version: '3.9'
       windows-3.5:
         imageName: "vs2017-win2016"

--- a/doc/CHANGELOG.rst
+++ b/doc/CHANGELOG.rst
@@ -5,6 +5,9 @@ This changelog *only* contains changes from the *first* pypi release (0.5.4.3) o
 
 Latest Changes:
 - **1.3.1_dev0 - 2021-06-05**
+
+  - Fixed issue with ssize_t on Windows for Python 3.10.
+
 - **1.3.0 - 2021-05-19**
 
   - Fixes for memory issues found when upgrading to Python 3.10 beta.

--- a/native/common/include/jp_booleantype.h
+++ b/native/common/include/jp_booleantype.h
@@ -84,7 +84,7 @@ public:
 	virtual void getView(JPArrayView& view) override;
 	virtual void releaseView(JPArrayView& view) override;
 	virtual const char* getBufferFormat() override;
-	virtual ssize_t getItemSize() override;
+	virtual Py_ssize_t getItemSize() override;
 	virtual void copyElements(JPJavaFrame &frame,
 			jarray a, jsize start, jsize len,
 			void* memory, int offset) override;

--- a/native/common/include/jp_bytetype.h
+++ b/native/common/include/jp_bytetype.h
@@ -87,7 +87,7 @@ public:
 	virtual void getView(JPArrayView& view) override;
 	virtual void releaseView(JPArrayView& view) override;
 	virtual const char* getBufferFormat() override;
-	virtual ssize_t getItemSize() override;
+	virtual Py_ssize_t getItemSize() override;
 	virtual void copyElements(JPJavaFrame &frame,
 			jarray a, jsize start, jsize len,
 			void* memory, int offset) override;

--- a/native/common/include/jp_chartype.h
+++ b/native/common/include/jp_chartype.h
@@ -80,7 +80,7 @@ public:
 	virtual void getView(JPArrayView& view) override;
 	virtual void releaseView(JPArrayView& view) override;
 	virtual const char* getBufferFormat() override;
-	virtual ssize_t getItemSize() override;
+	virtual Py_ssize_t getItemSize() override;
 	virtual void copyElements(JPJavaFrame &frame,
 			jarray a, jsize start, jsize len,
 			void* memory, int offset) override;

--- a/native/common/include/jp_doubletype.h
+++ b/native/common/include/jp_doubletype.h
@@ -84,7 +84,7 @@ public:
 	virtual void getView(JPArrayView& view) override;
 	virtual void releaseView(JPArrayView& view) override;
 	virtual const char* getBufferFormat() override;
-	virtual ssize_t getItemSize() override;
+	virtual Py_ssize_t getItemSize() override;
 	virtual void copyElements(JPJavaFrame &frame,
 			jarray a, jsize start, jsize len,
 			void* memory, int offset) override;

--- a/native/common/include/jp_floattype.h
+++ b/native/common/include/jp_floattype.h
@@ -84,7 +84,7 @@ public:
 	virtual void getView(JPArrayView& view) override;
 	virtual void releaseView(JPArrayView& view) override;
 	virtual const char* getBufferFormat() override;
-	virtual ssize_t getItemSize() override;
+	virtual Py_ssize_t getItemSize() override;
 	virtual void copyElements(JPJavaFrame &frame,
 			jarray a, jsize start, jsize len,
 			void* memory, int offset) override;

--- a/native/common/include/jp_inttype.h
+++ b/native/common/include/jp_inttype.h
@@ -89,7 +89,7 @@ public:
 	virtual void getView(JPArrayView& view) override;
 	virtual void releaseView(JPArrayView& view) override;
 	virtual const char* getBufferFormat() override;
-	virtual ssize_t getItemSize() override;
+	virtual Py_ssize_t getItemSize() override;
 	virtual void copyElements(JPJavaFrame &frame,
 			jarray a, jsize start, jsize len,
 			void* memory, int offset) override;

--- a/native/common/include/jp_longtype.h
+++ b/native/common/include/jp_longtype.h
@@ -88,7 +88,7 @@ public:
 	virtual void getView(JPArrayView& view) override;
 	virtual void releaseView(JPArrayView& view) override;
 	virtual const char* getBufferFormat() override;
-	virtual ssize_t getItemSize() override;
+	virtual Py_ssize_t getItemSize() override;
 	virtual void copyElements(JPJavaFrame &frame,
 			jarray a, jsize start, jsize len,
 			void* memory, int offset) override;

--- a/native/common/include/jp_primitivetype.h
+++ b/native/common/include/jp_primitivetype.h
@@ -41,7 +41,7 @@ public:
 	virtual void getView(JPArrayView& view) = 0;
 	virtual void releaseView(JPArrayView& view) = 0;
 	virtual const char* getBufferFormat() = 0;
-	virtual ssize_t getItemSize() = 0;
+	virtual Py_ssize_t getItemSize() = 0;
 	virtual void copyElements(JPJavaFrame &frame,
 			jarray a, jsize start, jsize len,
 			void* memory, int offset) = 0;

--- a/native/common/include/jp_shorttype.h
+++ b/native/common/include/jp_shorttype.h
@@ -88,7 +88,7 @@ public:
 	virtual void getView(JPArrayView& view) override;
 	virtual void releaseView(JPArrayView& view) override;
 	virtual const char* getBufferFormat() override;
-	virtual ssize_t getItemSize() override;
+	virtual Py_ssize_t getItemSize() override;
 	virtual void copyElements(JPJavaFrame &frame,
 			jarray a, jsize start, jsize len,
 			void* memory, int offset) override;

--- a/native/common/include/jp_voidtype.h
+++ b/native/common/include/jp_voidtype.h
@@ -54,7 +54,7 @@ public:
 	virtual void getView(JPArrayView& view) override;
 	virtual void releaseView(JPArrayView& view) override;
 	virtual const char* getBufferFormat() override;
-	virtual ssize_t getItemSize() override;
+	virtual Py_ssize_t getItemSize() override;
 	virtual void copyElements(JPJavaFrame &frame,
 			jarray a, jsize start, jsize len,
 			void* memory, int offset) override;

--- a/native/common/jp_array.cpp
+++ b/native/common/jp_array.cpp
@@ -178,8 +178,8 @@ JPArrayView::JPArrayView(JPArray* array, jobject collection)
 	// Second element is the shape of the array from which we compute the
 	// memory size, the shape, and strides
 	int dims;
-	ssize_t itemsize;
-	ssize_t sz;
+	Py_ssize_t itemsize;
+	Py_ssize_t sz;
 	{
 		JPPrimitiveArrayAccessor<jintArray, jint*> accessor(frame, (jintArray) item1,
 				&JPJavaFrame::GetIntArrayElements, &JPJavaFrame::ReleaseIntArrayElements);

--- a/native/common/jp_booleantype.cpp
+++ b/native/common/jp_booleantype.cpp
@@ -333,7 +333,7 @@ const char* JPBooleanType::getBufferFormat()
 	return "?";
 }
 
-ssize_t JPBooleanType::getItemSize()
+Py_ssize_t JPBooleanType::getItemSize()
 {
 	return sizeof (jboolean);
 }

--- a/native/common/jp_bytetype.cpp
+++ b/native/common/jp_bytetype.cpp
@@ -271,7 +271,7 @@ const char* JPByteType::getBufferFormat()
 	return "b";
 }
 
-ssize_t JPByteType::getItemSize()
+Py_ssize_t JPByteType::getItemSize()
 {
 	return sizeof (jbyte);
 }

--- a/native/common/jp_chartype.cpp
+++ b/native/common/jp_chartype.cpp
@@ -274,7 +274,7 @@ const char* JPCharType::getBufferFormat()
 	return "H";
 }
 
-ssize_t JPCharType::getItemSize()
+Py_ssize_t JPCharType::getItemSize()
 {
 	return sizeof (jchar);
 }

--- a/native/common/jp_class.cpp
+++ b/native/common/jp_class.cpp
@@ -424,7 +424,6 @@ void JPClass::getConversionInfo(JPConversionInfo &info)
 	JP_TRACE_OUT;
 }
 
-
 //</editor-fold>
 //<editor-fold desc="hierarchy" defaultstate="collapsed">
 

--- a/native/common/jp_doubletype.cpp
+++ b/native/common/jp_doubletype.cpp
@@ -312,7 +312,7 @@ const char* JPDoubleType::getBufferFormat()
 	return "d";
 }
 
-ssize_t JPDoubleType::getItemSize()
+Py_ssize_t JPDoubleType::getItemSize()
 {
 	return sizeof (jdouble);
 }

--- a/native/common/jp_floattype.cpp
+++ b/native/common/jp_floattype.cpp
@@ -295,7 +295,7 @@ const char* JPFloatType::getBufferFormat()
 	return "f";
 }
 
-ssize_t JPFloatType::getItemSize()
+Py_ssize_t JPFloatType::getItemSize()
 {
 	return sizeof (jfloat);
 }

--- a/native/common/jp_gc.cpp
+++ b/native/common/jp_gc.cpp
@@ -234,9 +234,9 @@ void JPGarbageCollection::onEnd()
 		}
 
 		// Predict if we will cross the limit soon.
-		ssize_t pred = current + 2 * (current - last);
+		Py_ssize_t pred = current + 2 * (current - last);
 		last = current;
-		if ((ssize_t) pred > (ssize_t) limit)
+		if ((Py_ssize_t) pred > (Py_ssize_t) limit)
 			run_gc = 2;
 
 		//		printf("consider gc %d (%ld, %ld, %ld, %ld) %ld\n", run_gc,

--- a/native/common/jp_inttype.cpp
+++ b/native/common/jp_inttype.cpp
@@ -298,7 +298,7 @@ const char* JPIntType::getBufferFormat()
 	return "i";
 }
 
-ssize_t JPIntType::getItemSize()
+Py_ssize_t JPIntType::getItemSize()
 {
 	return sizeof (jfloat);
 }

--- a/native/common/jp_longtype.cpp
+++ b/native/common/jp_longtype.cpp
@@ -297,7 +297,7 @@ const char* JPLongType::getBufferFormat()
 	return "q";
 }
 
-ssize_t JPLongType::getItemSize()
+Py_ssize_t JPLongType::getItemSize()
 {
 	return sizeof (jlong);
 }

--- a/native/common/jp_shorttype.cpp
+++ b/native/common/jp_shorttype.cpp
@@ -294,7 +294,7 @@ const char* JPShortType::getBufferFormat()
 	return "h";
 }
 
-ssize_t JPShortType::getItemSize()
+Py_ssize_t JPShortType::getItemSize()
 {
 	return sizeof (jshort);
 }

--- a/native/common/jp_voidtype.cpp
+++ b/native/common/jp_voidtype.cpp
@@ -122,7 +122,7 @@ const char* JPVoidType::getBufferFormat()
 	return NULL;
 }
 
-ssize_t JPVoidType::getItemSize()
+Py_ssize_t JPVoidType::getItemSize()
 {
 	return 0;
 }

--- a/native/python/include/jp_pythontypes.h
+++ b/native/python/include/jp_pythontypes.h
@@ -281,7 +281,7 @@ public:
 		return m_Contents.size();
 	}
 
-	PyObject* operator[](ssize_t i)
+	PyObject* operator[](Py_ssize_t i)
 	{
 		return m_Contents[i].get();
 	}

